crates/accelerate/src/two_qubit_decompose.rs

```diff
@@ -1739,6 +1718,152 @@ impl TwoQubitBasisDecomposer {
         })
     }
 
+    fn call_inner(
+        &self,
+        unitary: ArrayView2<Complex64>,
+        basis_fidelity: Option<f64>,
+        approximate: bool,
+        _num_basis_uses: Option<u8>,
+    ) -> PyResult<TwoQubitGateSequence> {
+        let basis_fidelity = if !approximate {
+            1.0
+        } else {
+            basis_fidelity.unwrap_or(self.basis_fidelity)
+        };
+        let target_decomposed =
+            TwoQubitWeylDecomposition::new_inner(unitary, Some(DEFAULT_FIDELITY), None)?;
+        let traces = self.traces(&target_decomposed);
+        let best_nbasis = traces
+            .into_iter()
+            .enumerate()
+            .map(|(idx, trace)| (idx, trace.trace_to_fid() * basis_fidelity.powi(idx as i32)))
+            .min_by(|(_idx1, fid1), (_idx2, fid2)| fid2.partial_cmp(fid1).unwrap())
+            .unwrap()
+            .0;
+        let best_nbasis = _num_basis_uses.unwrap_or(best_nbasis as u8);
```

Comment:
Can you calculate `best_nbasis` here in the block of an `unwrap_or_else` instead of using `unwrap_or`?

Reply:
Yeah I can do that, it is a potential optimization. Although realistically the hot path here is `best_nbasis` that private flag `_num_basis_uses`  I think is only used in tests.

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/12563/commits/655ce1d677d0f66265b8dd9a1e0631c31f2c26f9